openwisp_monitoring/device/api/views.py

```diff
@@ -203,6 +205,74 @@ def _write(self, request, pk):
                 metric.write(client['mac'])
             if created:
                 self._create_clients_graph(metric)
+        if 'resources' not in data:
+            return
+        self._write_cpu(data, pk, ct)
+        self._write_disk(data, pk, ct)
+        self._write_memory(data, pk, ct)
+
+    def _write_cpu(self, data, primary_key, content_type):
+        if 'load' in data['resources']:
```

Comment:
Improvement: same logic as we did already:

```
if 'load' not in data['resources']
    return
...
```

Reply:
An extra test was required after this change to maintain the coverage.

Reply:
Well, tbh I think we can assume that if 'resources' is present, then 'load', 'memory' and 'disk' are too. WDYT?

Reply:
Yes, it's safe unless the user modifies the `lua` script (unlikely, I think). So, shall I remove the lines verifying that `load`, `memory`, `disk` are present in `resources`?

Reply:
Yes please, let's take the opportunity to reduce the complexity :+1: